src/openms/source/FORMAT/MzTab.cpp

```diff
@@ -2598,21 +2598,29 @@ Not sure how to handle these:
     {
       prot_ids_.push_back(&i);
     }
+
+    // pre-reserve to prevent reallocations
+    size_t new_size = peptide_ids_.size();
+    for (Size i = 0; i < consensus_map.size(); ++i) new_size += consensus_map[i].getPeptideIdentifications().size();
+    if (export_unassigned_ids)
+    {
+      new_size += consensus_map.getUnassignedPeptideIdentifications().size();
+    }
+    peptide_ids_.reserve(new_size);
+
  
     // extract mapped IDs
     for (Size i = 0; i < consensus_map.size(); ++i)
     {
       const ConsensusFeature& c = consensus_map[i];
       const vector<PeptideIdentification>& p = c.getPeptideIdentifications();
-      peptide_ids_.reserve(peptide_ids_.size() + p.size());
```

Comment:
this was a true pessimization. Not sure how much preallocation really helps.

Reply:
I think we are speaking about gigabytes of data in our recent example. Doubling gigabytes led to the program not finding consecutive memory chunks. I was only involved in the discussion tho. @timosachsenberg knows more

Reply:
But yeah, seeing this: this will make this dataset work but a slightly bigger one will fail again, so we have to decide if this is worth it.

Reply:
`std::vector<const PeptideIdentification*> peptide_ids_;`

This vector should not take much space (compared to the actual data). If you run out of virtual mem, the reason is probably somewhere else (holding all elements in memory - this goes for both IDs and the mzTab data, for example; or a machine with not a lot of RAM).

Reply:
Yeah it is a bit strange. But if I break into the process it was always during the reserve. Not sure if ASAN or valgrind would ever finish though… the data after protein quantification ist not an issue. Only  takes about 70gb in memory.

Reply:
in the old code, reserve() is called as often as push_back() (same order of magnitude), but the former has runtime O(N), whereas push_back is O(1). So when stopping the process, you are likely to end up in reserve().
Why it runs out of virtual memory when doing a lot of new/delete would be interesting to know...

Reply:
I was suspecting some memory fragmentation.